tuner/tuner/libtuner.py

```diff
@@ -236,20 +237,59 @@ class ParsedDisptachBenchmarkResult:
 class IREEBenchmarkResult:
     # Default format follows output of iree-benchmark-module
     candidate_id: int
-    result_str: str
+    result_json: list
 
     def get_mean_time(self) -> Optional[float]:
-        if not self.result_str:
-            return None
-        pattern = r"process_time/real_time_mean\s+([\d.]+)\s\w{2}"
-        match = re.search(pattern, self.result_str)
-        if not match:
-            return None
-        try:
-            return float(match.group(1))
-        except ValueError:
+        """Compute the mean time for all of the benchmarks - The result will be in ms"""
+        if not self.result_json:
             return None
 
+        total_time = 0.0
+        count = 0
+
+        for benchmark in self.result_json:
+            real_time = benchmark.get("real_time")
+            time_unit = benchmark.get("time_unit", "ms")
+
+            if real_time is not None:
+                try:
+                    real_time_in_ms = self.convert_time_to_miliseconds(
+                        real_time, time_unit
+                    )
+                    total_time += real_time_in_ms
+                    count += 1
+                except ValueError as e:
+                    handle_error(
+                        condition=True,
+                        msg=f"Invalid benchmark JSON: {e}",
+                        error_type=ValueError,
+                        exit_program=True,
+                    )
+
+        if count > 0:
+            return total_time / count
+
+        return None
```

Comment:
We don't have to keep track of the count:
```suggestion
       if not self.result_json:
         return None
         
       return total_time_ms / len(self.result_json)
```